staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go

```diff
@@ -188,6 +192,15 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 		delegate:  delegateHandler,
 	}
 	establishingController := establish.NewEstablishingController(s.Informers.Apiextensions().V1().CustomResourceDefinitions(), crdClient.ApiextensionsV1())
+	var sc storageversion.Client
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.StorageVersionAPI) &&
```

Comment:
nit: do we even need these feature gate checks? Wouldn't it be enough to do that inside the handler?

Reply:
had the feature gates merely because `kubernetes.NewForConfig` could in theory return an error, and I don't want the apiextensions apiserver to fail start unexpectedly